appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.kt

```diff
@@ -591,18 +642,20 @@ abstract class AppIntroBase : AppCompatActivity(), AppIntroViewPager.OnNextPageR
     // Returns true if a permission has been requested
     private fun checkAndRequestPermissions(): Boolean {
```

Comment:
Please rewrite this function a bit. I see a couple of code smells here:

* `checkAndRequest...` you can maybe split it in 2 functions?
* Having a top level `return` plus a `return` inside one of the branch